.github/workflows/build-docker.yml

```diff
@@ -0,0 +1,103 @@
+name: Build and Publish Docker Images
+
+on:
+  push:
+    branches: [main]
+    tags:
+      - 'v*'
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
+
+permissions:
+  contents: read
+  packages: write
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_OWNER: ${{ github.repository_owner }}
+  REPO_NAME: ${{ github.event.repository.name }}
+  GATEWAY_IMAGE: gateway
+  FRONTEND_IMAGE: frontend
+  BACKEND_IMAGE: backend
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Log in to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract short SHA
+        id: vars
+        run: echo "SHORT_SHA=$(git rev-parse --short HEAD)" >> $GITHUB_OUTPUT
+        
+      - name: Install libxml2-utils
+        run: sudo apt-get update && sudo apt-get install -y libxml2-utils
+
+      - name: Extract RELEASE_VERSION from pom.xml
+        id: extract_version
+        run: |
+          VERSION=$(xmllint --xpath "/*[local-name()='project']/*[local-name()='version']/text()" pom.xml)
```

Comment:
While this works, it may be better to use something like `mvn help:evaluate -Dexpression=project.version -q -DforceStdout` or `mvn -q -Dexec.executable=echo -Dexec.args='${project.version}' --non-recursive exe
c:exec` as those also handle weird things like projects that derive their version number from a Maven property.